apis/v1beta1/config_types.go

```diff
@@ -86,6 +92,47 @@ type ConfigSpec struct {
 	Extensions           Extensions           `json:"extensions,omitempty"`
 }
 
+// OverrideForBuiltIn returns the override for the built-in extension with the
+// given name. If no override is found an empty string is returned.
+func (c ConfigSpec) OverrideForBuiltIn(bi string) string {
+	for _, ext := range c.UnsupportedOverrides.BuiltInExtensions {
+		if ext.Name != bi {
+			continue
+		}
+		return ext.Values
+	}
+	return ""
+}
+
+// ApplyEndUserAddOnOverrides applies the end-user provided addon config on top
+// of the provided addon configuration (cfg).
+func (c *ConfigSpec) ApplyEndUserAddOnOverrides(name, cfg string) (string, error) {
+	patch := c.OverrideForBuiltIn(name)
+	if len(cfg) == 0 || len(patch) == 0 {
+		if len(cfg) == 0 {
+			return patch, nil
+		}
+		return cfg, nil
+	}
```

Comment:
might be more readable this way (unless i'm missing a case)
```suggestion
	if len(cfg) == 0 {
	        return patch, nil
        }
	if len(patch) == 0 {
		return cfg, nil
	}
```